src/EventListener/BulkSavePropertyListener.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flagbit\Bundle\CategoryBundle\EventListener;
+
+use Akeneo\Pim\Enrichment\Component\Category\Model\CategoryInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Flagbit\Bundle\CategoryBundle\Entity\CategoryProperty;
+use Flagbit\Bundle\CategoryBundle\Repository\CategoryPropertyRepository;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+use function count;
+
+/**
+ * Category post save listener that handles bulk saves with properties.
+ */
+class BulkSavePropertyListener
+{
+    /** @var ParameterBag<string, array<string, mixed>> */
+    private ParameterBag $propertiesBag;
+    private CategoryPropertyRepository $repository;
+    private EntityManagerInterface $entityManager;
+
+    /**
+     * @phpstan-param ParameterBag<string, array<string, mixed>> $propertiesBag
+     */
+    public function __construct(
+        ParameterBag $propertiesBag,
+        CategoryPropertyRepository $repository,
+        EntityManagerInterface $entityManager
+    ) {
+        $this->propertiesBag = $propertiesBag;
+        $this->repository    = $repository;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @phpstan-param GenericEvent<mixed> $event
+     */
+    public function onBulkCategoryPostSave(GenericEvent $event): void
+    {
+        // Save category properties
+        foreach ($event->getSubject() as $category) {
+            if (! $category instanceof CategoryInterface) {
+                continue;
+            }
+
+            if (! $this->propertiesBag->has($category->getCode())) {
+                continue;
+            }
+
+            $properties = $this->propertiesBag->get($category->getCode());
+            if (count($properties) === 0) {
+                return;
+            }
+
+            $categoryProperty = $this->findProperty($category);
+            $categoryProperty->setProperties($properties);
+
+            $this->entityManager->persist($categoryProperty);
+            $this->entityManager->flush();
+        }
+    }
+
+    private function findProperty(CategoryInterface $category): CategoryProperty
+    {
+        /** @phpstan-var CategoryProperty|null $categoryProperty */
+        $categoryProperty = $this->repository->findOneBy(['category' => $category]);
+        if ($categoryProperty === null) {
+            $categoryProperty = new CategoryProperty($category);
+        }
+
+        return $categoryProperty;
+    }
```

Comment:
I'd suggest that we move this to `\Flagbit\Bundle\CategoryBundle\Repository\CategoryPropertyRepository` or a dedicated service and remove it from both event listeners that use the function to prevent the redundant definition of this function. 
Your oppinion @flagbird ?

Reply:
Let's postpose the suggested change to prevent an increase to the PR's scope. It can be done after it got merged.